frontend/src/metabase/hooks/use-confirm-leave-modal.ts

```diff
@@ -0,0 +1,78 @@
+import type { Location } from "history";
+import { useCallback, useEffect, useState } from "react";
+import type { InjectedRouter, Route } from "react-router";
+import { push } from "react-router-redux";
+
+import useBeforeUnload from "metabase/hooks/use-before-unload";
+import { useDispatch } from "metabase/lib/redux";
+
+interface UseConfirmLeaveModalInput {
+  router: InjectedRouter;
+  route: Route;
+  isEnabled: boolean;
+  isLocationAllowed?: (location: Location | undefined) => boolean;
+}
+
+interface UseConfirmLeaveModalResult {
+  opened: boolean;
+  close: () => void;
+  confirm: () => void;
+  nextLocation: Location | undefined;
+}
+
+export const IS_LOCATION_ALLOWED = (location?: Location) => {
+  /**
+   * If there is no "location" then it's beforeunload event, which is
+   * handled by useBeforeUnload hook - no reason to duplicate its work.
+   */
+  if (!location) {
+    return true;
+  }
+
+  return false;
```

Comment:
Would `return !location` work or is that less clear?

Reply:
that's a great point haha, this was copy-pasta